core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/handlers/internalresponse/EthSignTransactionResultProvider.java

```diff
@@ -0,0 +1,118 @@
+import tech.pegasys.web3signer.core.service.jsonrpc.JsonRpcRequest;
+import tech.pegasys.web3signer.core.service.jsonrpc.exceptions.JsonRpcException;
+import tech.pegasys.web3signer.core.service.jsonrpc.handlers.ResultProvider;
+import tech.pegasys.web3signer.core.service.jsonrpc.handlers.sendtransaction.transaction.EthTransaction;
+import tech.pegasys.web3signer.core.service.jsonrpc.handlers.sendtransaction.transaction.Transaction;
+import tech.pegasys.web3signer.core.service.jsonrpc.handlers.signing.TransactionSerializer;
+import tech.pegasys.web3signer.signing.ArtifactSignerProvider;
+
+import java.util.List;
+
+import io.vertx.core.json.DecodeException;
+import io.vertx.core.json.JsonObject;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class EthSignTransactionResultProvider implements ResultProvider<String> {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final long chainId;
+  private final ArtifactSignerProvider signerProvider;
+  private final JsonDecoder decoder;
+
+  public EthSignTransactionResultProvider(
+      final long chainId, final ArtifactSignerProvider signerProvider, final JsonDecoder decoder) {
+    this.chainId = chainId;
+    this.signerProvider = signerProvider;
+    this.decoder = decoder;
+  }
+
+  @Override
+  public String createResponseResult(final JsonRpcRequest request) {
+    LOG.debug("Transforming request {}, {}", request.getId(), request.getMethod());
+    final EthSendTransactionJsonParameters ethSendTransactionJsonParameters;
+    final Transaction transaction;
+    try {
+      ethSendTransactionJsonParameters =
+          fromRpcRequestToJsonParam(EthSendTransactionJsonParameters.class, request);
+      transaction = createTransaction(request, ethSendTransactionJsonParameters);
+
+    } catch (final NumberFormatException e) {
+      LOG.debug("Parsing values failed for request: {}", request.getParams(), e);
+      throw new JsonRpcException(INVALID_PARAMS);
+    } catch (final IllegalArgumentException | DecodeException e) {
+      LOG.debug("JSON Deserialization failed for request: {}", request.getParams(), e);
+      throw new JsonRpcException(INVALID_PARAMS);
+    }
+
+    if (!transaction.isNonceUserSpecified()) {
+      LOG.debug("Nonce not present in request {}", request.getId());
+      throw new JsonRpcException(INVALID_PARAMS);
+    }
+
+    LOG.debug("Obtaining signer for {}", transaction.sender());
+    try {
+      final TransactionSerializer transactionSerializer =
+          getTransactionSerializer(signerProvider, transaction.sender());
+      return transactionSerializer.serialize(transaction);
+    } catch (Exception e) {
+      LOG.debug("From address ({}) does not match any available account", transaction.sender());
+      throw new JsonRpcException(SIGNING_FROM_IS_NOT_AN_UNLOCKED_ACCOUNT);
+    }
+  }
+
+  private TransactionSerializer getTransactionSerializer(
```

Comment:
nit: don't really need a separate method for this now that we always create a `TransactionSerializer`

Reply:
Done